runtime/src/main/java/org/corfudb/runtime/LogReplicationClient.java

```diff
@@ -0,0 +1,277 @@
+                            return false;
+                        }
+
+                        clientDestinations = clientDestinations.toBuilder().addDestinationIds(destination).build();
+                    }
+                    else {
+                        clientDestinations = SinksInfoVal.newBuilder().addDestinationIds(destination).build();
+                    }
+
+                    txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinations, null);
+                    txn.commit();
+
+                    return true;
+                }
+                catch (Exception e) {
+                    log.error("Unable to add destination", e);
+                    return false;
+                }
+            }).setOptions(x -> x.setMaxRetryThreshold(Duration.ofSeconds(SYNC_THRESHOLD))).run();
+        } catch (InterruptedException e) {
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Add a list destination to LogReplicationSourceMetadataTable for monitoring
+     *
+     * @param logicalGroup String representative of the group of tables to be replicated
+     * @param remoteDestinations Clusters which the logicalGroup should be replicated to
+     * @return true if all destinations were newly registered
+     */
+    public boolean addDestination(String logicalGroup, List<String> remoteDestinations) {
+        if (Strings.isNullOrEmpty(logicalGroup)) {
+            log.error("Invalid logicalGroup");
+            return false;
+        }
+        try {
+            return IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientInfoKey clientInfoKey = ClientInfoKey.newBuilder()
+                            .setClientName(clientName)
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    SinksInfoVal clientDestinations;
+
+                    boolean hasNullOrEmptyOrDuplicates =
+                            remoteDestinations.stream().anyMatch(Objects::isNull) ||
+                            remoteDestinations.stream().anyMatch(String::isEmpty) ||
+                            remoteDestinations.stream().distinct().count() < remoteDestinations.size();
+
+                    if (hasNullOrEmptyOrDuplicates) {
+                        log.error("Invalid destination(s) found in list");
+                        return false;
+                    }
+
+                    if (txn.isExists(LR_SOURCE_METADATA_TABLE_NAME, clientInfoKey)) {
+                        clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                        for (String dest : remoteDestinations) {
+                            if (clientDestinations.getDestinationIdsList().contains(dest)) {
```

Comment:
The list could have some existing destinations along with new ones.
I think we should handle it by finding out the new destinations first and then write to the table.

we can have this check if we. don't find any new destinations to add. 
Also, I think we shouldn't return "false" in that case as that could translate to destinations not being ack'ed by LR which would not be the case.
We can just log a msg for debugging, maybe DEBUG level, and return true